components/legend/legend.service.js

```diff
@@ -283,51 +283,54 @@ export default function (HsUtilsService) {
         HsUtilsService.instOf(layer.getSource(), TileWMS) ||
         HsUtilsService.instOf(layer.getSource(), ImageWMS)
       ) {
-        const subLayerLegends = layer.getSource().getParams().LAYERS.split(',');
-        for (let i = 0; i < subLayerLegends.length; i++) {
-          subLayerLegends[i] = me.getLegendUrl(
-            layer.getSource(),
-            subLayerLegends[i],
-            layer
-          );
+        const sourceParamLayers = layer.getSource().getParams().LAYERS;
+        if (angular.isDefined(sourceParamLayers)) {
+          const subLayerLegends = sourceParamLayers.split(',');
+          for (let i = 0; i < subLayerLegends.length; i++) {
+            subLayerLegends[i] = me.getLegendUrl(
+              layer.getSource(),
+              subLayerLegends[i],
```

Comment:
While you are touching the code, this might be nicer:
```suggestion
          for (const subLayerLegend of subLayerLegends) {
            subLayerLegend = me.getLegendUrl(
              layer.getSource(),
              subLayerLegend,
```